hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -78,10 +78,10 @@ object Elaborator:
       case OuterCtx.Function(sym) => S(sym)
       case _ => parent.flatMap(_.getNonLocalRetHandler)
     // Returns N if no return handler is required (a direct `return` is possible).
-    // Returns S(N) if the return handler is required but not found.
+    // Returns S(N) if there is no function in scope.
```

Comment:
A function in scope would mean that there's a sbiling `fun` statement in the current block or in an outer block.

```suggestion
    // Returns S(N) if we are not in a function scope.
```